### DISCUSSION TITLE
xmake 在 nixos 中的默认工具链无法正常链接

### DISCUSSION THREAD

**Discussion by yangyagami** [xmake-io/xmake#6019] [Q&A] [answered]
xmake 在 nixos 上编译：
![screenshot](https://github.com/user-attachments/assets/882b5cd4-7e17-4b26-8836-300d48560b8f)

**Comment by waruqi** [comment, marked as answer]
估计是被系统环境的 $LD = ld 给影响了吧。 你可以去了 LD env 试试，我这边没 nixos 环境

**Comment by Freed-Wu** [comment]
Refer: https://github.com/Freed-Wu/rime.nvim/blob/main/shell.nix#L22-L25

https://github.com/NixOS/nixpkgs/issues/314313#issuecomment-2134252094

**Comment by waruqi** [comment]
试下这个 patch  https://github.com/xmake-io/xmake/pull/6022

**Reply by yangyagami** [reply]
![screenshot](https://github.com/user-attachments/assets/59aff8b9-f1e5-4b9e-a053-b74d6dd8761c)
我尝试从源码构建，构建完成后，运行没有反应。

**Reply by yangyagami** [reply]
```bash
# 我是这样构建的
cd xmake
./configure
make -j4
```

**Reply by waruqi** [reply]
只需要 xmake update -s github:xmake-io/xmake#envs 更新就行了

**Reply by yangyagami** [reply]
感谢作者的回复，刚刚试着更新 dev 分支也不行，目前 nixos 下看来无法用传统方法去使用。
```bash
[yangsiyu@nixos:~/下载]$ xmake update -s github:xmake-io/xmake#dev
update version dev from https://github.com/xmake-io/xmake.git ..
  => download https://github.com/xmake-io/xmake.git .. ok
  => install script to /nix/store/mvvbw9sk0z40193j69ddmpvgkwx0s447-xmake-2.9.6/serror: run `sh /nix/store/mvvbw9sk0z40193j69ddmpvgkwx0s447-xmake-2.9.6/share/xmake/scripts/update-script.sh /nix/store/mvvbw9sk0z40193j69ddmpvgkwx0s447-xmake-2.9.6/share/xmake /tmp/.xmake1000/250103/xmakesrc/dev/xmake` failed, may permission denied!
note: try continue to run `sh /nix/store/mvvbw9sk0z40193j69ddmpvgkwx0s447-xmake-2.9.6/share/xmake/scripts/update-script.sh /nix/store/mvvbw9sk0z40193j69ddmpvgkwx0s447-xmake-2.9.6/share/xmake /tmp/.xmake1000/250103/xmakesrc/dev/xmake` with administrator permission again (pass -y or --confirm=y/n/d to skip confirm)?
please input: y (y/n)
y
[sudo] yangsiyu 的密码：
failed

```

**Reply by Freed-Wu** [reply]
> and it seems to be your build is failed with exit value 255.

Fail is due to `/usr/share/xmake` cannot be found. Try `DESTDIR=/tmp/xxx make install`
